chain/events/events_called.go

```diff
@@ -516,47 +465,32 @@ func (me *messageEvents) checkNewCalls(ts *types.TipSet) (map[triggerID][]eventD
 		}
 	})
 
-	return res, nil
+	return res
 }
 
 // Get the messages in a tipset
 func (me *messageEvents) messagesForTs(ts *types.TipSet, consume func(*types.Message)) {
 	seen := map[cid.Cid]struct{}{}
 
-	for _, tsb := range ts.Blocks() {
-		me.blockMsgLk.Lock()
-		msgsI, ok := me.blockMsgCache.Get(tsb.Cid())
-		var err error
-		if !ok {
-			msgsI, err = me.cs.ChainGetBlockMessages(context.TODO(), tsb.Cid())
-			if err != nil {
-				log.Errorf("messagesForTs MessagesForBlock failed (ts.H=%d, Bcid:%s, B.Mcid:%s): %s", ts.Height(), tsb.Cid(), tsb.Messages, err)
-				// this is quite bad, but probably better than missing all the other updates
-				me.blockMsgLk.Unlock()
-				continue
-			}
-			me.blockMsgCache.Add(tsb.Cid(), msgsI)
+	for i, tsb := range ts.Cids() {
+		msgs, err := me.cs.ChainGetBlockMessages(context.TODO(), tsb)
```

Comment:
get message block by block. but in fact there are many messages not to be executed or duplicate. could use ChainGetMessagesInTipset here?

Reply:
That's a good point, but something for a followup patch. I'm replicating the current behavior here.

Reply:
Hm. Actually, I _think_ this code is correct as-is. We _want_ to know if a message was included in the tipset, even if it wasn't executed. That way we can check to see if the message successfully applied. Otherwise, we'll keep waiting for the message to show up even though it never will.

Any context here @magik6k?

Reply:
I think this is fine for now as it doesn't seem change the current behavior